Pr/enhancements

aljones15 opened the pull request.

mattcollier commented on .eslintrc.js:
```diff
@@ -1,7 +1,8 @@
+  env: {
+    browser: true
+  },
+  plugins: ["jsdoc"],
```
this produces error that strings must be single quoted in other files, but not here?  What's up with that?

Also, extra blank line at bottom.

aljones15 replied:
good question actually because veres-wallet's eslint does run on .eslint so not entirely sure. made the corrections.

mattcollier commented on AccountService.js:
```diff
@@ -4,7 +4,7 @@
 
 import axios from 'axios';
-
+const headers = {Accept: 'application/ld+json, application/json'};
```
I still don't know if we need to override the default headers.  @gannan08 @dlongley

aljones15 replied:
good point we should seek a professional opinion also Content-Type should contain application/json-patch+json on the update method.

EDIT: update method now uses application/json-patch+json

dlongley replied:
Well, we do want to accept `application/ld+json` so it seems right to me. Why wouldn't it be?

aljones15 replied:
I tested this this morning and the default content type if application/json
we are not returning linked data yet, but not sure if we are in the future.
so @dlongley should I remove the application/ld+json Accept type or keep it because accounts might be linked data in the future?

dlongley commented on AccountService.js:
```diff
@@ -27,17 +27,76 @@ export class AccountService {
-    const response = await axios.post(url, {email}, {
-      headers: {'Accept': 'application/ld+json, application/json'}
-    });
+  async create({baseUrl = this.config.urls.base, email}) {
```
Using `baseUrl` here is a bit weird -- it makes sense in the other cases because it's understood that the other parameters will be mixed with it. But that's not true here. I'm hesitant to request this be changed back to `url` since all of the other calls use `baseUrl`... but that's really the only reason. We can leave it as is and fix it up later if it's a real point of confusion.

aljones15 replied:
no issue fixing it

dlongley commented:
`implementation` should be considered a dead branch -- it's only there because it's being used by some demos. This should be merged into `dev` which we want to get merged into master after we get more review on everything there ... and then delete it.

mattcollier commented:
@dlongley so you would like to have a common practice of overriding the default axios headers for code in  this layer?  That's fine, I just want to understand the convention.

aljones15 commented:
> `implementation` should be considered a dead branch -- it's only there because it's being used by some demos. This should be merged into `dev` which we want to get merged into master after we get more review on everything there ... and then delete it.

ok so basically here is what happened:

Put in a pull request yesterday for veres-wallet
another PR for veres-wallet was approved.
I rebased that PR in dev into my branch
this changes bedrock-web-account to use implementation
because we had decided on using implementation over dev
I thought that the npm pointing at implementation is correct
hence I put in a PR to push this to implementation.
so anyways if this is good let's make a release on this and bedrock-account-http
and then in my PR for veres-wallet I can update bedrock-web-account and bedrock-account-http to point to new releases.


@dlongley updated to move into master will try to squash it a little and will make sure the change log has been updated.

squashed down to 14, removing redundant commit messages, change log is good.
